src/services/publishing/publishing.ts

```diff
@@ -4,31 +4,38 @@ import { signDocument, SUPPORTED_SIGNING_ALGORITHM } from "@govtechsg/open-attes
 import { TitleEscrowCreatorFactory, TradeTrustErc721Factory } from "@govtechsg/token-registry";
 import { TitleEscrowCreator } from "@govtechsg/token-registry/types/TitleEscrowCreator";
 import { providers, Signer, Wallet } from "ethers";
+import { Provider } from "@ethersproject/abstract-provider";
 import { getGsnRelaySigner } from "../../common/config/decrypt";
-import { PublishingJob, WrappedDocument } from "../../types";
+import { ConnectedSigner, PublishingJob, WrappedDocument } from "../../types";
 import { supportsInterface } from "./utils";
 
-const assertAddressIsSmartContract = async (address: string, wallet: Wallet): Promise<void> => {
-  const code = await wallet.provider.getCode(address);
+const assertAddressIsSmartContract = async (address: string, provider: Provider): Promise<void> => {
```

Comment:
why change the interface? refactoring?

Reply:
Cause the KMS Signer return ethers.signer, it won't be able to parse into Wallet properties. The function only requires providers thus I refactor the argument to Provider

Reply:
👌